rules/alerting.go

```diff
@@ -338,8 +338,22 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 		return nil, err
 	}
 
-	r.mtx.Lock()
```

Comment:
This is not right as we are mutating alerts outside of the lock.

Reply:
Thank you , How to do it with right way?

Reply:
Updated.